UnityProject/Assets/Scripts/Items/Botany/Fruit/NoFruit.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using Mirror;
+using NaughtyAttributes;
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+namespace Items.Botany.Fruit
+{
+	public class NoFruit : NetworkBehaviour, ICheckedInteractable<HandApply>
+	{
+
+		[Serializable]
+		class ObjectToSpawn
+		{
+			public GameObject fruit;
+			public SpriteDataSO fruitSprite;
+		}
+
+		[SyncVar] private bool isReadyToBeHit = false;
+		private SpriteHandler handler;
+		private int currentIndex = 0;
+
+		[SerializeField] private List<ObjectToSpawn> chanceToSpawn = new List<ObjectToSpawn>();
+		[SerializeField] private GameObject facehugger;
+
+
+		private void Awake()
+		{
+			handler = GetComponentInChildren<SpriteHandler>();
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			if (isReadyToBeHit == false && interaction.IsAltClick) return false;
+			if (isReadyToBeHit && interaction.Intent != Intent.Harm) return false;
+			return true;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (isReadyToBeHit == false)
+			{
+				isReadyToBeHit = true;
+				Chat.AddExamineMsg(interaction.Performer, "You pull the tip of the nofruit and watch as multiple colors flash in front of you!");
+				StartCoroutine(CycleSprites());
+				return;
+			}
+
+			if (interaction.UsedObject == null || interaction.UsedObject.Item().Size < ItemSize.Medium)
+			{
+				var hasObject = interaction.UsedObject ? "a more" : "a";
+				Chat.AddExamineMsg(interaction.Performer,
+					$"You need {hasObject} blunt object to hit this!");
+				return;
+			}
+			StopCoroutine(CycleSprites());
+			if (DMMath.Prob(1f))
+			{
+				Spawn.ServerPrefab(facehugger, gameObject.AssumedWorldPosServer());
+				return;
+			}
+			else
+			{
+				Spawn.ServerPrefab(chanceToSpawn[currentIndex].fruit, gameObject.AssumedWorldPosServer());
+			}
+			_ = Despawn.ServerSingle(gameObject);
+		}
+
+		private IEnumerator CycleSprites()
+		{
+			while (isReadyToBeHit || this != null)
+			{
+				handler.SetSpriteSO(chanceToSpawn[currentIndex].fruitSprite);
+				yield return WaitFor.Seconds(0.3f);
+				currentIndex += 1;
```

Comment:
```suggestion
				currentIndex = (currentIndex + 1) % chanceToSpawn.Count;
```
then you don't need the if